### PR TITLE
HUE-9574 [core] Apply timeout to thrift-over-http calls

### PR DESCRIPTION
**Pull request by wing2fly**

### CODE REVIEW

File: desktop/core/src/desktop/lib/rest/http_client.py [hunk 1]
```diff
@@ -199,7 +199,11 @@ def execute(self, http_method, path, params=None, data=None, headers=None, allow
         self.logger.warn("GET and DELETE methods do not pass any data. Path '%s'" % path)
         data = None
 
+    if headers and 'timeout' in headers:
+      timeout = int(headers['timeout'])
+
     request_kwargs = {'allow_redirects': allow_redirects, 'timeout': timeout}
+    LOG.debug("timeout for rest call: %d" % timeout)
```
Comment (romainr) [inline desktop/core/src/desktop/lib/rest/http_client.py:206]: rest --> REST

**Review by romainr — APPROVED** [review]
Ok, so we override the timeout value if present in the headers.

Maybe worth 2 quick unit tests btw?

File: desktop/core/src/desktop/lib/rest/http_client.py [hunk 2]
```diff
@@ -199,7 +200,11 @@ def execute(self, http_method, path, params=None, data=None, headers=None, allow
         self.logger.warn("GET and DELETE methods do not pass any data. Path '%s'" % path)
         data = None
 
+    if headers and 'timeout' in headers:
+      timeout = int(headers['timeout'])
+
```
Comment (romainr) [inline desktop/core/src/desktop/lib/rest/http_client.py:205]: PS: should we move the LOG.debug here instead? 

'Overriding timeout xxx via header value xx'?

Comment (wing2fly) [reply desktop/core/src/desktop/lib/rest/http_client.py:206]: Fixed

**Pull request merged**